QuestionBot/QuestionBot/Model/InMemoryStore.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace QuestionBot.Model {
+
+    public class InMemoryStore : IStore {
+        private IList<IRecord> allRecords = new List<IRecord>();
+        private static int _questionId = 0;
+
+        public IRecord CreateRecord(string question)
+        {
+            _questionId++;
+
+            Record newQuestion = new Record(_questionId, question, DateTime.Now);
+            allRecords.Add(newQuestion);
+
+            return newQuestion;
+        }
+
+        public IRecord UpdateRecord(int id, string answer)
+        {
+            IRecord recordToUpdate = allRecords.FirstOrDefault(recordItem => recordItem.ID == id);
```

Comment:
Ah so you are correct in that `FirstOrDefault` will return `null` instead of throwing... but then two lines down when you do  `recordToUpdate.Answer = answer;` this will throw if `recordToUpdate` is `null`. But this would have been caught by your automated tests so no worries :) I think using try/catch is the right call here.